lib/netsuite/support/records.rb

```diff
@@ -35,6 +35,12 @@ def to_attributes!(hash, kname, v)
           hash[:attributes!][kname]['externalId'] = v.external_id
         end
 
+        if v.respond_to?(:replace_all) && !v.replace_all.to_s.empty?
+          hash[:attributes!] ||= {}
+          hash[:attributes!][kname] ||= {}
+          hash[:attributes!][kname]['replaceAll'] = v.replace_all
+        end
```

Comment:
@AgustinSuarez why did this functionality need to be moved out of the sublist module? I believe `replaceAll` is only applicable to sublists, so I don't think it should be required anywhere else. What am I missing here?

Reply:
So in order to upsert sales orders and other type of transaction correctly,  you need to be able to define it as follows  `<listRel:addressbookList replaceAll="false"/>` The problem with the previous code is that it added it as a field and didn't take it into consideration. 

This helps when upserting records like addresses and line items